src/util/getReceiverAddress.ts

```diff
@@ -46,15 +46,14 @@ const getReceiverAddress = ({ targetChainDetails, sourceChainDetails, receiverAd
   };
 
   try {
-    const [validatedDerivedAcccount, rest] = checkAddress(receiverAddress, targetSS58Format);
+    const decodedReceiverAddress = base58Decode(receiverAddress);
+    const [isValidDerivedAcccount, , , ss58Decoded] = checkAddressChecksum(decodedReceiverAddress);
 
-    if (validatedDerivedAcccount) {
-      return { address: receiverAddress, formatFound: targetChain };
-    }
+    const formatFound = getChainBySS58Prefix(ss58Decoded) || ss58Decoded;
 
-    const parts = rest?.split(',');
-    const prefix = parts![2].split(' ');
-    const formatFound = getChainBySS58Prefix(prefix[2]) || prefix[2];
+    if (isValidDerivedAcccount) {
+      return { address: receiverAddress, formatFound };
+    }
```

Comment:
I think this is where the issue with tests is. Note that previously `validatedDerivedAccount` as only `true` if `targetSS58Format` was matching the decoded format.
Currently I think we will get `true` if the address was decoded successfully, but we should compare the `ss58Decoded` format with the expected one and re-format if needed.

Reply:
@adamzerella how's everything going with this comment? do you need some help to address it? I think we are very close :)

Reply:
Makes sense to me guys! Just trying to find the time to update the PR, aiming to push something later this week 🤞🏻